lib/activerecord-import/adapters/postgresql_adapter.rb

```diff
@@ -30,6 +31,92 @@ def post_sql_statements( table_name, options ) # :nodoc:
     end
   end
 
+  # Add a column to be updated on duplicate key update
+  def add_column_for_on_duplicate_key_update( column, options={} ) # :nodoc:
+    arg = options[:on_duplicate_key_update]
+    if arg.is_a?( Hash )
+      columns = arg.fetch(:columns) { arg[:columns] = [] }
+      case columns
+      when Array then columns << column.to_sym unless columns.include?( column.to_sym )
+      when Hash then columns[column.to_sym] = column.to_sym
+      end
+    elsif arg.is_a?( Array )
+      arg << column.to_sym unless arg.include?( column.to_sym )
+    end
+  end
+
+  # Returns a generated ON CONFLICT DO UPDATE statement given the passed
+  # in +args+.
+  def sql_for_on_duplicate_key_update( table_name, *args ) # :nodoc:
+    arg = args.first
+    if arg.is_a?( Array ) || arg.is_a?( String )
+      arg = { :columns => arg }
+    end
+    return unless arg.is_a?( Hash )
+
+    constraint_name = arg[ :constraint_name ]
+    conflict_target = arg[ :conflict_target ]
+    columns         = arg.fetch( :columns, [] )
+
+    sql = " ON CONFLICT "
+    conflict_target = sql_for_conflict_target( conflict_target, constraint_name )
+
+    if columns.respond_to?( :empty? ) && columns.empty?
```

Comment:
@jkowens said:
> Yes, but the code currently looks to see if the table has updated_on or updated_at timestamp columns and if it does those are automatically added to the list of columns, so it's unlikely columns will be empty.

While likely true it's not guaranteed so I agree with @triptec that we should have a test for this.

> I'm thinking this code was put in place for MySQL since its on duplicate key update clause does not have a "do nothing" option, so columns were required

MySQL has INSERT IGNORE which allows I believe achieve the same end-result: ignore inserting rows that fail a required constraint check. This is supported in activerecord-import by passing the `ignore: true` option to `import(...)`. 

In hindsight, this is a terribly named option in activerecord-import.

> That brings up a good question, since Postgres does have a "DO NOTHING" option, should we not automatically add the update columns?

I think we add them if the caller passed in other columns, but only then. See below `on_duplicate_key_ignore` suggestion. 

@triptec said:
> I've been trying to do something like this:
Category.import categories, :on_duplicate_key_update=>{ :conflict_target => [:name], :columns => [], validate: false }

@triptec, so you don't really want to update anything, you just want to ignore duplicate records, yes? If so, what about something like this (which doesn't exist yet):

```
Category.import categories, :on_duplicate_key_ignore=>{ :conflict_target => [:name] }, validate: false
```

This would make it clear that you intend to ignore duplicates and make help us cleanly separate this from the case where we automatically update timestamps when present.